helm-chart/crds/meshconfig.yaml

```diff
@@ -0,0 +1,182 @@
+        properties:
+          apiVersion:
+            description: 'APIVersion defines the versioned schema of this representation
+              of an object. Servers should convert recognized schemas to the latest
+              internal value, and may reject unrecognized values. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#resources'
+            type: string
+          kind:
+            description: 'Kind is a string value representing the REST resource this
+              object represents. Servers may infer this from the endpoint the client
+              submits requests to. Cannot be updated. In CamelCase. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#types-kinds'
+            type: string
+          metadata:
+            type: object
+          spec:
+            description: Spec defines the desired configuration for NGINX Service Mesh.
+            properties:
+              meshConfigClassName:
+                description: MeshConfigClassName used for this MeshConfig. This is
+                  the name of a MeshConfigClass resource.
+                minLength: 1
+                type: string
+              accessControlMode:
+                description: AccessControlMode for service-to-service communication.
+                enum:
+                - allow
+                - deny
+                type: string
+              clientMaxBodySize:
+                description: ClientMaxBodySize is NGINX client max body size.
+                pattern: ^\d+[kKmMgG]?$
+                type: string
+              disableAutoInjection:
+                description: DisableAutoInjection disables automatic sidecar injection globally.
+                type: boolean
+              enabledNamespaces:
+                description: EnabledNamespaces is the list of namespaces where automatic
+                  sidecar injection is enabled.
+                items:
+                  type: string
+                type: array
+              mtls:
+                description: Mtls is the configuration for mutual TLS.
+                properties:
+                  caKeyType:
+                    description: CaKeyType is the key type used for the SPIRE Server
+                      CA.
+                    enum:
+                    - ec-p256
+                    - ec-p384
+                    - rsa-2048
+                    - rsa-4096
+                    type: string
+                  caTTL:
+                    description: CaTTL is the CA/signing key TTL in hours(h). Min
+                      value 24h. Max value 999999h.
+                    pattern: ^1[0-9]{2,5}(h)|2[4-9](h)|2[0-9][0-9]{1,5}(h)|[3-9][0-9]{1,5}(h)$
+                    type: string
+                  mode:
+                    description: Mode for pod-to-pod communication.
+                    enum:
+                    - "off"
```

Comment:
Is there a reason off is inside quotation marks?

Reply:
When generating the CRD, I think it's a keyword, which is why the quotes were added. But let me just remove them and see if it has any effect.

Reply:
In Yaml v1.1, on/off are considered boolean values. Yaml v1.2 removed this stipulation, but for backwards compatibility it's probably best to just leave it quoted to ensure that it's interpreted as a string value.